src/wallet/wallet.cpp

```diff
@@ -4010,6 +4010,11 @@ bool CWallet::ParameterInteraction()
         LogPrintf("%s: parameter interaction: -salvagewallet=1 -> setting -rescan=1\n", __func__);
     }
 
+    // -zapwallettx implies dropping the mempool on startup
+    if (GetBoolArg("-zapwallettxes", false) && SoftSetBoolArg("-persistmempool", false)) {
```

Comment:
Yes, you're right - that is strange. I just copied the existing log below.

I've fixed them both so they actually print the value of zapwallettxes.

Oh, and surprise! zapwallettxes isn't a bool!